src/hooks/sdk/useMediaStreamMetrics.tsx

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useState, useEffect } from 'react';
+import { ClientMetricReport } from 'amazon-chime-sdk-js';
+
+import { useAudioVideo } from '../../providers/AudioVideoProvider';
+
+function isValidMetric(metric: any) {
+  return typeof metric === 'number' && !Number.isNaN(metric);
+}
+
+interface MediaStreamMetrics {
+  audioPacketsSentFractionLossPercent: number | null; // Percentage of audio packets lost (1s) from client to server
+  audioPacketsReceivedFractionLossPercent: number | null; // Percentage of audio packets lost from server to client
+  availableOutgoingBandwidth: number | null;
+  availableIncomingBandwidth: number | null;
+  videoStreamMetrics: { [key: string]: { [key: string]: {[key: string]: number} } }
+}
+
+export function useMediaStreamMetrics() {
+  const audioVideo = useAudioVideo();
+  const [mediaStreamMetrics, setMetrics] = useState<MediaStreamMetrics>({
+    audioPacketsSentFractionLossPercent: null,
+    audioPacketsReceivedFractionLossPercent: null,
+    availableOutgoingBandwidth: null,
+    availableIncomingBandwidth: null,
+    videoStreamMetrics: {},
+  });
+
+  useEffect(() => {
+    if (!audioVideo) {
+      return;
+    }
+
+    const observer = {
+      metricsDidReceive(clientMetricReport: ClientMetricReport): void {
+        const metricReport = clientMetricReport.getObservableMetrics();
+
+        let videoStreamMetrics = {};
+        let availableOutgoingBandwidth = 0;
+        let availableIncomingBandwidth = 0;
+        let audioPacketsSentFractionLossPercent = 0;
+        let audioPacketsReceivedFractionLossPercent = 0;
+
+        if (isValidMetric(metricReport.audioPacketLossPercent)) {
```

Comment:
You can use object destructuring in this function which will help you avoid writing `metricReport.` few times.

```js
const { audioPacketLossPercent, availableSendBandwidth .. } = metricReport;
```

Reply:
Good point. Will do.